src/faces.jl

```diff
@@ -0,0 +1,122 @@
+# Create recipe: 
+@Makie.recipe(Faces, grid) do scene
+    Makie.Attributes(;
+    
+      # Use default theme:
+      Makie.default_theme(scene, Makie.Mesh)...,
+      Makie.default_theme(scene, Makie.Lines)...,
+      Makie.default_theme(scene, Makie.Scatter)...,
+
+      # Custom arguments:
+      color      = :pink,
+      colormap   = :bluesreds,
+      colorrange = nothing,
+      fieldstyle = :nodes,
+
+      # Otherwise, 2D plots don't use custom default colors:
+      cycle      = nothing,
+
+      # Force no shading:
+      shading    = false
+    )
+end
+  
+# Plot! especialization:  
+function Makie.plot!(plot::Faces{<:Tuple{Grid}})
+
+  grid = plot[:grid][]
+  Dc = num_cell_dims(grid)
+
+  # Dispatch edges according to cell dimension Dc:
+  if Dc == 0
+    plot_Ngon_vertices!(plot, grid)
+  elseif Dc == 1
+    plot_Ngon_edges!(plot, grid)
+  else
+    plot_Ngon_faces!(plot, grid |> to_face_grid)
+  end
+
+end
+
+# 2D plots:
+function plot_Ngon_faces!(plot::Faces{<:Tuple{Grid}}, grid::Grid)
+
+  # Retrieve plot attributes:
+  fieldstyle    = plot[:fieldstyle][]
+  color         = plot[:color][]
+
+  grid_ = to_plot_mesh(grid)
+  if color isa AbstractVector
+    if plot[:colorrange][] === nothing
+      plot[:colorrange][] = extrema(color)
+    end
+    if length(color) != GeometryBasics.coordinates(grid_) |> length
+      color = if fieldstyle == :nodes
+                to_dg_node_values(grid, color)
+              elseif fieldstyle == :cells
+                to_dg_cell_values(grid, color)
+              else
+                error("Invalid fieldstyle")
+              end
+    end
+  end
+  Makie.mesh!(plot, grid_; 
```

Comment:
Hi, @paurierap, I think I am commenting an outdated code... but are you still passing all the list of attributes when you interface to `Makie.mesh!`, `Makie.linesegments!` and `Makie.scatter!`, as done here for `Makie.mesh!`?

Reply:
Hi, @ericneiva, I am still passing a list of attributes. If you know a better way to do it, let me know!

Reply:
I just wonder if you can use function ‘with_theme’ to avoid passing all attributes: http://makie.juliaplots.org/stable/theming.html

Reply:
Good point! I will give it a look to see if it fits!